src/invariant/ConservationOfLumens.cpp

```diff
@@ -71,7 +79,54 @@ calculateDeltaBalance(LedgerEntry const* current, LedgerEntry const* previous)
     }
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
     case CONTRACT_DATA:
-        break;
+    {
+        auto const& contractData = current ? current->data.contractData()
+                                           : previous->data.contractData();
+        if (contractData.contractID != lumenContractID ||
+            contractData.key.type() != SCV_VEC || !contractData.key.vec() ||
+            contractData.key.vec().size() == 0)
+        {
+            return 0;
+        }
+
+        // The balanceSymbol should be the first entry in the SCVec
+        if (!(contractData.key.vec()->at(0) == balanceSymbol))
+        {
+            return 0;
+        }
+
+        auto getAmount = [&amountSymbol](LedgerEntry const* entry) -> int64_t {
+            if (!entry)
+            {
+                return 0;
+            }
+
+            // The amount should be the first entry in the SCMap
+            auto const& val = entry->data.contractData().val;
+            if (val.type() == SCV_MAP && val.map() && val.map()->size() != 0)
```

Comment:
Same in here, perhaps need to add an assert statement on the expected data type and size.